src/olympia/discovery/admin.py

```diff
@@ -81,7 +83,19 @@ class Media:
     raw_id_fields = ('addon',)
     readonly_fields = ('recommended_status', 'previews',)
     view_on_site = False
-    list_select_related = ('primaryhero',)
+
+    def get_queryset(self, request):
+        qset = (
+            DiscoveryItem.objects.all()
+            .select_related('primaryhero')
+            .prefetch_related(
+                Prefetch(
+                    'addon',
+                    queryset=Addon.unfiltered.all().only_translations()),
+                Prefetch(
+                    'addon___current_version',
+                    queryset=Version.unfiltered.all().only_translations())))
```

Comment:
Actually... if we want to push this further... I think you can remove that second prefetch, and add `select_related('_current_version')` on the `Addon` queryset that is being prefetched above. It will prevent versions transformers from running, including the translations one, but we don't need translations on `Version` here. Worth a try locally.

Reply:
Also in any case, we should add a comment for that method explaining that we're forced to use `prefetch_related()` because of translations, and that we use `only_translations()` because we don't care about the other transforms.

Reply:
Fixed in 3a791c7 and 6dfdbe3